docs/root/configuration/http/http_conn_man/vhds.rst

```diff
@@ -9,6 +9,100 @@ will call to dynamically fetch
 a name and set of domains that get routed to it based on the incoming request's
 host header.
 
+By default in RDS, all routes for a cluster are sent to every Envoy instance
+in the mesh. This causes scaling issues as the size of the cluster grows. The
+majority of this complexity can be found in the virtual host configurations, of
+which most are not needed by any individual proxy. 
+
+In order to fix this issue, the Virtual Host Discovery Service (VHDS) protocol
+uses the delta xDS protocol to allow a route configuration to be subscribed to
+and the necessary virtual hosts to be requested as needed. Instead of sending
+all virtual hosts with a route config, using VHDS will allow an Envoy instance
+to subscribe and unsubscribe from a list of virtual hosts stored internally in
+the xDS management server. The xDS management server will monitor this list and
+use it to filter the configuration sent to an individual Envoy instance to only
+contain the subscribed virtual hosts.
+
+Virtual Host Resource Naming Convention
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Virtual hosts in VHDS are identified by a combination of the name of the route
+configuration to which the virtual host belong as well as the host HTTP "host"
+header (authority for HTTP2) entry. Resources should be named as follows:
+
+<route configuration name>/<host entry>
+
+Note that matching should be done from right to left since a host entry cannot
+contain slashes while a route configuration name can. 
+
+Joining/Reconnecting to a management server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When an Envoy instance forms a delta xDS connection to the xDS management
+server for the first time, it will receive a base configuration filtered down
+to a subset of routes that are likely useful to the Envoy instance. For
+example, these might be filtered down by routes that exist in the same
+application. The xDS management server will send this same collection of
+resources on reconnect.
+
+.. figure:: diagrams/delta_rds_connection.svg
+   :alt: Delta RDS connection
+
+Subscribing to Resources
+^^^^^^^^^^^^^^^^^^^^^^^^
+VHDS allows resources to be :ref:`subscribed <xds_protocol_delta_subscribe>` to
+using a :ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with the 
+:ref:`type_url <envoy_api_field_DeltaDiscoveryRequest.type_url>` set to 
+`type.googleapis.com/envoy.api.v2.route.VirtualHost` 
+and :ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>` 
+set to a list of virtual host resource names for which it would like
+configuration. 
+
+If a route for the contents of a host/authority header cannot be resolved,
+the active stream is paused while a :ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` is sent. 
+When a :ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>` is
+recieved where the host/authority header values exactly matches the 
+:ref:`aliases <envoy_api_field_resource.aliases>`,
```

Comment:
The `name` and `aliases` fields will have values in the form `<route configuration name>/<host entry>`, right?  Those are the names that are used in the VHDS request, so that's what the resources have to be named in order to match.

I think the check here should be that the `name` or `aliases` fields match the route configuration name and host/authority header value in the format specified above.